docs/modeling/jointfitter.rst

```diff
@@ -0,0 +1,189 @@
+       def input_units(self):
+           if self.mean.unit is None:
+               return None
+           return {self.inputs[0]: self.mean.unit}
+
+Data to be fit is simulated.  The 1st spectral segment will have a spectral
+resolution that is a factor of 2 higher than the second segment.  The first
+segment will have wavelengths from 1 to 6 and the second from 4 to 10 giving
+an overlapping wavelength region from 4 to 6.
+
+.. code-block:: python
+
+   # Generate fake data
+   mean = 5.1
+   sigma1 = 0.2
+   sigma2 = 0.4
+   noise = 0.10
+
+   # compute the central amplitudes so the lines in each segement have the
+   # same area
+   area = 1.5
+   amp1 = area / np.sqrt(2.0 * math.pi * sigma1 ** 2)
+   amp2 = area / np.sqrt(2.0 * math.pi * sigma2 ** 2)
+
+   # segment 1
+   np.random.seed(0)
+   x1 = np.linspace(1.0, 6.0, 200)
+   y1 = amp1 * np.exp(-0.5 * (x1 - mean) ** 2 / sigma1 ** 2)
+   y1 += np.random.normal(0.0, noise, x1.shape)
+
+   # segment 2
+   np.random.seed(0)
+   x2 = np.linspace(4.0, 10.0, 200)
+   y2 = amp2 * np.exp(-0.5 * (x2 - mean) ** 2 / sigma2 ** 2)
+   y2 += np.random.normal(0.0, noise, x2.shape)
+
+Now define the models to be fit and fitter to use.  Then fit the two simulated
+datasets.
+
+.. code-block:: python
+
+   # define the two models to be fit
+   gjf1 = AreaGaussian1D(
+       area=1.0, mean=5.0, stddev=1.0, bounds={"area": (0, 10), "stddev": (0.1, 0.3)}
+   )
+   gjf2 = AreaGaussian1D(
+       area=1.0, mean=5.0, stddev=1.0, bounds={"area": (0, 10), "stddev": (0.1, 0.5)}
+   )
+
+   # define the jointfitter specifying the parameters in common and their initial values
+   fit_joint = fitting.JointFitter(
+       [gjf1, gjf2], {gjf1: ["area", "mean"], gjf2: ["area", "mean"]}, [1.0, 5.0]
+   )
+
+The resulting fit parameters show that the area and mean wavelength of the
+two AreaGaussian1D models are exactly the same while the width (stddev) is
+different reflecting the different spectral resolutions of the two segments.
+
+.. code-block:: python
+
+   print("AreaGaussian1 parameters")
+   print(gjf1.param_names)
+   print(gjf1.parameters)
+   print("AreaGaussian1 parameters")
+   print(gjf1.param_names)
+   print(gjf2.parameters)
+
+::
+
+   AreaGaussian1 parameters
+   ('area', 'mean', 'stddev')
+   [1.48697226 5.09826068 0.19761087]
+   AreaGaussian1 parameters
+   ('area', 'mean', 'stddev')
+   [1.48697226 5.09826068 0.4015368 ]
+
+The simulated data and best fit models can be plotted showing good agreement
+between the two AreaGaussian1D models and the two spectral segments.
+
+.. code-block:: python
+
+   # Plot the data with the best-fit models
+   plt.figure(figsize=(8,5))
+   plt.plot(x1, y1, 'bo', alpha=0.25)
+   plt.plot(x2, y2, 'go', alpha=0.25)
+   plt.plot(x1, gjf1(x1), 'b--', label='AreaGaussian1')
+   plt.plot(x2, gjf2(x2), 'g--', label='AreaGaussian2')
+   plt.xlabel('Wavelength')
+   plt.ylabel('Flux')
+   plt.legend(loc=2)
+
+.. image:: example_spectal_segments_jointfitter.png
```

Comment:
Why provide a static image when you can have matplotlib sphinx plot directive to do it for you?